spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -621,19 +638,20 @@ private File retrieveOriginalFileFromHeader(Message<?> message) {
 		return null;
 	}
 
-	private File handleFileMessage(final File sourceFile, File tempFile, final File resultFile) throws IOException {
+	private File handleFileMessage(final File sourceFile, File tempFile, final File resultFile, final Message<?> requestMessage)
+			throws IOException {
 		if (!FileExistsMode.APPEND.equals(this.fileExistsMode) && this.deleteSourceFiles) {
```

Comment:
Looks like we missed `APPEND_NO_FLUSH` here (original bug - not related to this PR).

Reply:
Good catch!
But looks like it has to be addressed in the separate issue/PR...